https://stepik.org/lesson/573098/step/9

А теперь будем уменьшать параметр $β$. Пусть теперь $β=0.5$. Подсчитайте значение $F_β$ для `precision=0.7` `recall=0.8`, а потом для случая `precision=0.8` `recall=0.7`.

У вас получится в итоге два числа. В ответ как и раньше напишите **максимальное из них**.

In [ ]:
# max((1 + .5**2) * .7 * .8 / (.5**2 * .7 + .8), (1 + .5**2) * .8 * .7 / (.5**2 * .8 + .7))
max((1 + beta**2) * precision * recall / (beta**2 * precision + recall) \
    for beta, precision, recall in [[.5, .7, .8], [.5, .8, .7]])

0.7777777777777778

# Решение

$$
F_{\beta} = \frac {(1 + \beta^2) \cdot precision \cdot recall}{\beta^2 \cdot precision + recall}
$$

$$~$$

$$
\frac {(1 + 0.5^2) \cdot 0.7 \cdot 0.8}{0.5^2 \cdot 0.7 + 0.8} = 0.7179
$$

$$~$$

$$
\frac {(1 + 0.5^2) \cdot 0.8 \cdot 0.7}{0.5^2 \cdot 0.8 + 0.7} = 0.7778
$$